src/main/java/org/broadinstitute/hellbender/tools/copynumber/arguments/CopyNumberArgumentValidationUtils.java

```diff
@@ -84,30 +90,74 @@ public static boolean isSameDictionary(final SAMSequenceDictionary dictionary1,
     }
 
     /**
-     * Resolve intervals from an {@link IntervalArgumentCollection} and a read-count file.
+     * Resolve intervals from an {@link IntervalArgumentCollection} and a read-count path.
      * If intervals are not specified in the {@link IntervalArgumentCollection}, they are taken from the
-     * read-count file.  The sequence dictionary is taken from the read-count file.  A {@link SimpleIntervalCollection}
+     * read-count path.  The sequence dictionary is taken from the read-count path.  A {@link SimpleIntervalCollection}
      * constructed using these intervals and sequence dictionary is returned and can be used for further validation.
      */
-    public static SimpleIntervalCollection resolveIntervals(final File readCountFile,
+    public static SimpleIntervalCollection resolveIntervals(final String readCountPath,
                                                             final IntervalArgumentCollection intervalArgumentCollection,
                                                             final Logger logger) {
-        IOUtils.canReadFile(readCountFile);
+        Utils.nonNull(readCountPath);
+        IOUtils.assertFileIsReadable(IOUtils.getPath(readCountPath));
         Utils.nonNull(intervalArgumentCollection);
         Utils.nonNull(logger);
-        final SimpleCountCollection firstReadCounts = SimpleCountCollection.read(readCountFile);
-        final SAMSequenceDictionary sequenceDictionary = firstReadCounts.getMetadata().getSequenceDictionary();
-        final LocatableMetadata metadata = new SimpleLocatableMetadata(sequenceDictionary);
 
         if (intervalArgumentCollection.intervalsSpecified()) {
             logger.info("Intervals specified...");
             validateIntervalArgumentCollection(intervalArgumentCollection);
-            return new SimpleIntervalCollection(metadata,
-                    intervalArgumentCollection.getIntervals(sequenceDictionary));
         } else {
-            logger.info(String.format("Retrieving intervals from read-count file (%s)...", readCountFile));
-            return new SimpleIntervalCollection(metadata, firstReadCounts.getIntervals());
+            logger.info(String.format("Retrieving intervals from read-count file (%s)...", readCountPath));
         }
+
+        final SimpleCountCollection readCounts = BucketUtils.isCloudStorageUrl(readCountPath)
+                ? SimpleCountCollection.readFromGCS(readCountPath)
+                : SimpleCountCollection.read(new File(readCountPath));
```

Comment:
I thought about this and agree it would be cleaner, but since we have to do the extra step of merging intervals in the GCS case, I think it's OK to encourage the caller to do this step externally (so we don't repeat this work in the loop where these subset methods are called, inexpensive as it may typically be).  Unfortunately, this is encouragement is only given via the method docs and not by any actual validation.

Reply:
I'm okay with this as long as there are tests for both code paths